server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -968,7 +970,7 @@ public void createShard(
         RecoveryState recoveryState = indexService.createRecoveryState(shardRouting, targetNode, sourceNode);
         IndexShard indexShard = indexService.createShard(shardRouting, globalCheckpointSyncer, retentionLeaseSyncer);
         indexShard.addShardFailureCallback(onShardFailure);
-        indexShard.startRecovery(
+        final CheckedRunnable<RuntimeException> recoveryRunnable = () -> indexShard.startRecovery(
```

Comment:
Using `RuntimeException` for `CheckedRunnable` doesn't seem right. Maybe just inline the `indexShard.startRecovery`?